File: visa-dev-samplecode-master/vdp-ruby/test/tns/visa_travel_notification_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/visa_api_client', __FILE__)

class VisaTravelNotificationServiceTest < Test::Unit::TestCase
  def setup
    @visa_api_client = VisaAPIClient.new
    @config = YAML.load_file('configuration.yml')
    @departureDate = Time.now.strftime("%Y-%m-%d")
    @returnDate = (Time.now + (7*24*60*60)).strftime("%Y-%m-%d")
    @travelNotificationRequest ='''{
    "addTravelItinerary": {
    "returnDate": "''' + @returnDate + '''",
    "departureDate": "''' + @departureDate +'''",
    "destinations": [
                      {
                        "state": "CA",
                        "country": "840"
                      }
                    ],
    "primaryAccountNumbers": ''' + @config['tnsCardNumbers'] + ''',
    "userId": "Rajesh",
    "partnerBid": "''' + @config['tnsPartnerBid'] + '''"
    }
   }'''
  end
 
  def test_addTravelItenary
    base_uri = 'travelnotificationservice/'
    resource_path = 'v1/travelnotification/itinerary'
    response_code = @visa_api_client.doMutualAuthRequest("#{base_uri}#{resource_path}", "Add Travel Itenary Test", "post", @travelNotificationRequest)
    assert_equal("200", response_code, "Add Travel Itenary test failed")
  end
end

File: visa-dev-samplecode-master/vdp-node/test/tns/visatravelnotificationtest.js
var request = require('request');
var fs = require('fs');
var config = require('../../config/configuration.json');
var VisaAPIClient = require('../../libs/visaapiclient.js');
var assert = require('chai').assert;
var randomstring = require('randomstring');

var req = request.defaults();
var userId = config.userId ;
var password = config.password;
var keyFile = config.key;
var certificateFile = config.cert;

describe('Visa Travel Notification Service Test', function() {
	var visaAPIClient = new VisaAPIClient();
	var departureDate = new Date();
	var returnDate = new Date();
	returnDate.setDate(returnDate.getDate() + 7);
	var travelNotificationRequest = JSON.stringify({
            "addTravelItinerary": {
            "returnDate" : returnDate.toISOString().replace(/T.*$/, ''),
            "departureDate" : departureDate.toISOString().replace(/T.*$/, ''),
            "destinations": [
                              {
                                "state": "CA",
                                "country": "840"
                              }
                            ],
            "primaryAccountNumbers": config.tnsCardNumbers,
            "userId": "Rajesh",
            "partnerBid": config.tnsPartnerBid
            }
        });
	
	it('Add Travel Itenary Test',function(done) {
		this.timeout(10000);
		var baseUri = 'travelnotificationservice/';
		var resourcePath = 'v1/travelnotification/itinerary';
		visaAPIClient.doMutualAuthRequest(baseUri + resourcePath, travelNotificationRequest, 'POST', {}, 
		function(err, responseCode) {
		    if(!err) {
		    	assert.equal(responseCode, 200);
		    } else {
		        assert(false);
		    }
		    done();
		});
	});
});
